aten/src/ATen/native/Convolution.cpp

```diff
@@ -500,8 +500,14 @@ struct ConvParams {
     if (needs_64bit_indexing_no_split(input, weight)) {
       return false;
     }
+    bool compiledWithMiOpen = false;
+    // Checking the pointer before using it
+    auto cudaHooksPtr = detail::getCUDAHooksPtr();
+    if(cudaHooksPtr != nullptr) {
+      compiledWithMiOpen = cudaHooksPtr->compiledWithMIOpen();
+    }
     return ((input.scalar_type() == at::kFloat) || (input.scalar_type() == at::kHalf) || (input.scalar_type() == at::kBFloat16))
-           && detail::getCUDAHooks().compiledWithMIOpen()
+           && compiledWithMiOpen
```

Comment:
A less invasive fix would be to just move this line down one condition? If input.is_cuda() is true, we're guaranteed that checking the cuda hooks is safe.

Reply:
sure